control-plane/agents/core/src/core/registry.rs

```diff
@@ -97,6 +99,16 @@ impl Registry {
         registry.init().await;
         registry
     }
+
+    /// Adds the default port to the endpoint if one isn't already specified.
+    fn format_store_endpoint(endpoint: &str) -> String {
```

Comment:
Can we use the more common approaches? For example std::net::SocketAddrV4? That makes it in the end a whole lot easier. 

```
use std::net::{SocketAddrV4, Ipv4Addr};
let socket = SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), 8080);
```

https://doc.rust-lang.org/std/net/struct.SocketAddrV4.html

Reply:
Thanks for the suggestion. I have changed this function to construct and return a SocketAddrV4 type.

Reply:
The changes that I made for this ended up breaking the bdd & rust tests, so I'll raise a separate work item for this.